auto-remediate/security_hub_rules.py

```diff
@@ -67,58 +67,62 @@ def iam_user_unused_credentials_check(self, resource_id):
+                        self.logging.error(f"Could not retrieve IAM Login Profile for User '{user_name}'.")
                         self.logging.error(sys.exc_info()[1])
-                        return False
-                
-                # check access keys usage
-                try:
-                    list_access_keys = client.list_access_keys(UserName=user_name)
-                except:
-                    self.logging.error(f"Could not list IAM Access Keys for User '{user_name}'.")
-                    self.logging.error(sys.exc_info()[1])
-                    return False
-                
-                for access_key in list_access_keys.get('AccessKeyMetadata'):
-                    access_key_id = access_key.get('AccessKeyId')
-                    access_key_date = access_key.get('CreateDate')
-                    access_key_status = access_key.get('Status')
-                    
-                    if access_key_status == 'Active':
-                        if SecurityHubRules.get_day_delta(access_key_date) > 90:
+                    else:
+                        login_profile_date = login_profile.get('LoginProfile').get('CreateDate')
+                        if SecurityHubRules.get_day_delta(login_profile_date) > 90:
                             try:
-                                client.delete_access_key(AccessKeyId=access_key_id)
-                                self.logging.info(f"Deleted IAM Access Key '{access_key_id}' for User '{user_name}'.")
+                                client.delete_login_profile(UserName=user_name)
+                                self.logging.info(f"Deleted IAM Login Profile for User '{user_name}'.")
                             except:
-                                self.logging.error(f"Could not delete IAM Access Key for User '{user_name}'.")
+                                self.logging.error(f"Could not delete IAM Login Profile for User '{user_name}'.")
                                 self.logging.error(sys.exc_info()[1])
                                 return False
-                
-                return True
+                    
+                    # check access keys usage
+                    try:
+                        list_access_keys = client.list_access_keys(UserName=user_name)
+                    except:
+                        self.logging.error(f"Could not list IAM Access Keys for User '{user_name}'.")
+                        self.logging.error(sys.exc_info()[1])
+                        return False
+                    
+                    for access_key in list_access_keys.get('AccessKeyMetadata'):
+                        access_key_id = access_key.get('AccessKeyId')
+                        access_key_date = access_key.get('CreateDate')
+                        access_key_status = access_key.get('Status')
+                        
+                        if access_key_status == 'Active':
```

Comment:
this can be just one level of if (combine is active check and 90 days check)

Reply:
d3ca679